tools/gdal/gdal_macros.xml

```diff
@@ -186,8 +186,14 @@
   </xml>
 
 
+  <xml name="gdal_param_co">
+      <param name="co" type="text" label="-co NAME=VALUE" help="e.g : COMPRESS=JPEG" value="" />
+  </xml>   
 
 
+  <xml name="gdal_param_ot">
+      <param name="ot" type="text" label="-ot Datatype" value="" />
```

Comment:
Same here, why not a list with all datatypes which we support?

Reply:
People might want to export their result files or use them in others gdal tools even if Galaxy don't recognise them ?

Reply:
maybe, but lets tackle these use-cases if they arise
Lets make the first version useable to people that do not know the gdal internals.

E.g. we could add a select list and one entry is "specify advanced parameters ..."

Reply:
I'm quite ok here with a select, the option can only have one of the following `{Byte/Int16/UInt16/UInt32/Int32/Float32/Float64/CInt16/CInt32/CFloat32/CFloat64}` even though I don't know which one are more Galaxy usable. 

But for option like -co where there are ~160 formats available and you can specify more than 1, I still think the user should write by hand what he wants. It is already an advanced parameter for skilled users that might want more than the ~10% that galaxy recognises.

That's a guess, I'm not a gdal user and I don't understand half of the parameters ..  :/

Reply:
True and unfortunately, "simple" GDAL workflows don't use these advanced parameters...

Reply:
ok :)